### ISSUE TITLE
[vllm] [2.12 regression] Qwen2.5-Math-PRM-7B reward logits diverge from golden (test_prm_models_with_golden_outputs)

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#181249]
## Summary

Under torch 2.12.0 + triton 3.7.0, vLLM's `test_prm_models_with_golden_outputs[half-Qwen/Qwen2.5-Math-PRM-7B]` fails because the reward-model (PRM) logits diverge beyond `torch.allclose(..., rtol=1.5e-2)` from the saved golden outputs:

> `assert False`
> `+  where False = torch.allclose(tensor([[6.3610e-04, 9.9951e-01], [8.1885e-01, 1.8127e-01], [2.5543e-02, 9.7461e-01], [4.3821e-04, 9.9951e-01]]),`
> `                                 tensor([[6.3748e-04, 9.9936e-01], ...]))`

The vLLM outputs and the pre-saved golden (captured on torch 2.11) differ in the 3rd-4th decimal place — small enough to look like numerical drift but consistent enough to break a 1.5% tolerance. Passes on torch 2.11. Blocking the torch 2.12 upgrade for vLLM (vllm-project/vllm#40077).

## Environment

- `torch`: 2.12.0+cu130 (test channel)
- `triton`: 3.7.0
- CUDA: 13.0
- Python: 3.12.13
- Model: `Qwen/Qwen2.5-Math-PRM-7B`, `dtype=half`, `max_model_len=1024`

## Reproduction

Failing test:
```
tests/models/language/pooling/test_reward.py::test_prm_models_with_golden_outputs[half-Qwen/Qwen2.5-Math-PRM-7B]
```

```python
assert torch.allclose(golden_output, vllm_output, 1.5e-2)
```

## Reproducibility

- torch-2.12 branch: failed https://buildkite.com/vllm/ci/builds/62583#019db793-6c92-4c23-9ebd-d84ecad165df
- main (torch 2.11) — passes on every recent build:
  - 2026-04-21 daily: https://buildkite.com/vllm/ci/builds/62383
  - 2026-04-22 nightly: https://buildkite.com/vllm/ci/builds/62456
  - 2026-04-22 daily: https://buildkite.com/vllm/ci/builds/62566
  - 2026-04-23 nightly: https://buildkite.com/vllm/ci/builds/62613

## Diagnosis request

Same family as pytorch/pytorch#181168 (qwen2_vl output divergence) and pytorch/pytorch#181169 (gsm8k accuracy drop): small, deterministic numerical drift on torch 2.12 that is large enough to break tests with `torch.allclose` tolerance or accuracy thresholds. Likely root cause is a kernel-level change (matmul / softmax / attention / rotary) from torch 2.12 or triton 3.7. Is this intentional behavior change or a regression?

## Links

- vLLM PR: vllm-project/vllm#40077
- Umbrella: pytorch/pytorch#180899

**Comment by angelayi** [comment]
This issue seems to pass after clearing the torch.compile cache, and after some claude bisection seems like this cherry pick should fix the issue -- https://github.com/pytorch/pytorch/issues/180506#issuecomment-4316729371

**Comment by atalman** [comment]
Not reproducing anymore on latest run [buildkite.com/vllm/ci/builds/63283](https://buildkite.com/vllm/ci/builds/63283)
